### DISCUSSION TITLE
Webcam image capture support

### DISCUSSION THREAD

**Discussion by veryphatic** [directus/directus#18997] [Feature Requests]
### Summary

The existing single image file and multiple file interfaces are great. The image editor is a handy feature when working with single images; and it would be great if this could be added to the multiple file interface 😉, but I digress. 

I'd like to request both these interfaces be extended with support to capture an image from the devices' web camera; either a built in webcam, or a USB webcam (both use the `navigator.mediaDevices` API, request the same permissions etc). 

### Motivation

I've been building a data collection system for a not for profit organisation. One of their requirements is to capture one or more images from a USB microscope camera (or mobile phone camera) and store the image as a data point in a collection. 

### Detailed Design

I originally looked at developing this feature into an extension, using both the `file-image` and `files` interface as a starting point. From the `file-image` interface take the image editor, and use the `files` interface to manage the images. However, the `files` interface manages the files in a M2M relationship, has a bunch of private view components and is deeply integrated into Directus; making it difficult to copy and paste, and subsequently build this in a Directus way.

I imagine, if it were being built into the core, it would be achieved with a simple `media device` component to:
- support requesting permissions;
- ability to reset the permission, eg. if you accidently deny and block camera access
- set and store camera selection using `localstorage`
- display the live image in a `video` element
- button to capture the image and emit the base64 blob (possibly some offscreen `canvas` processing to resize and/or optimise the image to be within a target size).
- show the captured image to use (or reset)

### Must Have:
- manage device permissions
- capture multiple images

### Should Have:
- reuse existing `files` interface

### Could Have:
- add image editor to `files` interface per row

### Won't Have:
- Video support
- Audio support

### Drawbacks

I don't feel there are many drawbacks, if any. The `mediaDevices` API is stable and doesn't require any core changes to Directus or the existing interface components mentioned. 

### Alternatives

As mentioned, the image support built into Directus is pretty great already; attempting to make this as an extension would result in doubling up on existing features and be more difficult than is necessary to manage the image-data relationships.

### Adoption Strategy

Not a breaking change; it would simply provide a new option to upload a file.

**Comment by veryphatic** [comment]
Bump 🤜🤛 for ❤️

**Comment by veryphatic** [comment]
Hi, me again👋 I've just released a webcam interface https://github.com/veryphatic/directus-webcam-capture-interface

I've put effort into ensuring it's compatible with the File List interface as well as incorporating some portable components and hooks.. Let me know what you think 😄

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
